LayoutTests/fast/canvas/canvas-createPattern-video-modify.html

```diff
@@ -21,31 +21,29 @@
     ];
 
     var video = document.createElement("video");
-    video.addEventListener("loadeddata", loadeddata);
     video.src = findMediaFile("video", "../../media/content/test");
+    Promise.all([ once(video, "loadeddata"), waitForVideoFrame(video) ]).then(firstframe);
 
-    function loadeddata()
+    async function firstframe()
     {
-        if (!modified) {
-            canvas = document.createElement("canvas");
-            canvas.width = 2.5 * video.videoWidth;
-            canvas.height = 2.5 * video.videoHeight;
-            document.body.appendChild(canvas);
-
-            context = canvas.getContext("2d");
-            context.fillStyle = context.createPattern(video, "repeat");
-
-            video.src = findMediaFile("video", "../../media/content/counting");
-            modified = !modified;
-        } else {
-            context.fillRect(0, 0, canvas.width, canvas.height);
-
-            expectedResults.forEach(function(element) {
-                checkPixels(context, element[0], element[1], element[2], element[3], element[4], videoCanvasPixelComparisonTolerance());
-            });
-
-            finishJSTest();
-        }
+        canvas = document.createElement("canvas");
+        canvas.width = 2.5 * video.videoWidth;
+        canvas.height = 2.5 * video.videoHeight;
+        document.body.appendChild(canvas);
+
+        context = canvas.getContext("2d");
+        context.fillStyle = context.createPattern(video, "repeat");
+
+        video.src = findMediaFile("video", "../../media/content/counting");
+
+        await waitForVideoFrame(video);
```

Comment:
This seems ok like this but we are waiting for loadeddata event in the other code paths.

Reply:
like we don't need to wait for loadeddata on the other.